certdb/ocspstapling/ocspstapling.go

```diff
@@ -0,0 +1,117 @@
+package ocspstapling
+
+import (
+	"crypto"
+	"crypto/x509"
+	"crypto/x509/pkix"
+	"encoding/asn1"
+	"encoding/base64"
+	"errors"
+	"fmt"
+	"github.com/cloudflare/cfssl/certdb"
+	cferr "github.com/cloudflare/cfssl/errors"
+	"github.com/cloudflare/cfssl/helpers"
+	"github.com/google/certificate-transparency/go"
+	"golang.org/x/crypto/ocsp"
+)
+
+// OID of the SCT extension
+var sctExtOid = asn1.ObjectIdentifier{1, 3, 6, 1, 4, 1, 11129, 2, 4, 2}
+
+// StapleSCTList inserts a list of Signed Certificate Timestamps into all OCSP
+// responses in a database wrapped by a given certdb.Accessor.
+//
+// NOTE: This function is patterned after the exported Sign method in
+// https://github.com/cloudflare/cfssl/blob/master/signer/local/local.go
+func StapleSCTList(acc certdb.Accessor, serial, aki string, scts []ct.SignedCertificateTimestamp, issuer *x509.Certificate, priv crypto.Signer) error {
+	// Grab all OCSP records that match serial and aki
+	ocspRecs, err := acc.GetOCSP(serial, aki)
+	if err != nil {
+		return err
+	}
+
+	if len(ocspRecs) == 0 {
+		// { OCSPRecord does not exist }
+		return cferr.Wrap(cferr.CertStoreError, cferr.RecordNotFound, errors.New("empty OCSPRecord"))
+	}
+
+	// Add the SCTs to each ocsp response
+	for _, rec := range ocspRecs {
+		der, err := base64.StdEncoding.DecodeString(rec.Body)
+		if err != nil {
+			return cferr.Wrap(cferr.CertificateError, cferr.DecodeFailed,
+				errors.New("failed to decode Base64-encoded OCSP response"))
+		}
+
+		response, err := ocsp.ParseResponse(der, nil)
+		if err != nil {
+			return cferr.Wrap(cferr.CertificateError, cferr.ParseFailed,
+				errors.New("failed to parse DER-encoded OCSP response"))
+		}
+
+		// Serialize the list of SCTs
```

Comment:
You can probably remove this --- the function that's called immediately below is called `SerializeSCTList`; unless it's doing something else, the comment isn't needed.